docs/integrator-guide/workflow-using-ecs.md

```diff
@@ -0,0 +1,89 @@
+
+1. In `cumulus-tf/main.tf`, set `cumulus.throttled_queues[0].execution_limit` to the maximum per minute rate, e.g., 25.
+2. In `cumulus-tf/additions.tf`, set `background_job_queue_watcher.input` attribute 
+  `messageLimit` to the same value as `cumulus.throttled_queues[0].execution_limit`.
+
+For example, the MAAP Project CMR configuration previously used an in-memory queue. When the queue was full, 500 status code responses started to be returned. With the memory available on that instance, if more than ~500 granules "backed up" on this queue, the CMR ingest would hang and require a restart. The configuration could handle 25-50 granules/min at a steady pace, but a "burst" of 750 messages in 3 minutes could cause it to hang.
+
+### Discover Granules and Queue Granules Performance
+
+Both the Discover Granules and Queue Granules ECS Tasks require enough memory to load and process all of the granules. If there is not enough available memory in the EC2 instance to which they are assigned, they may fail. In the initial stack deployment, this may appear as a repeated message like:
+
+```
+null_resource.restart_discover_granules_ecs_task[0]: Still creating... [13h35m41s elapsed]
+null_resource.restart_queue_granules_ecs_task[0]: Still creating... [13h35m41s elapsed]
+```
+
+The fix for this is to increase the memory settings for each of these tasks. These are the `memory_reservation` setting in the `discover_granules_service` module in `cumulus-tf/addtions.tf` and the `queue_granules_service` in `cumulus-tf/lambda_queue_granules.tf`.
+
+When making a change to an existing stack (usually to the task count), either of these task may simply fail to start with no error. It is important to check that they are both running after every change update.
+
+This happens when the SyncGranule tasks are allocated in a way to don't leave enough memory resources on a single node to run the DiscoverGranules or QueueGranules task. This can be fixed either by (1) in the AWS Console, update the SyncGranule Service definition to have a value of `1` for `Number of tasks` or (2) change `sync_granule_service.desired_count` to 1 and deploy. This creates a configuration with enough available resources to run DiscoverGranules and QueueGranules. Then, set `sync_granule_service.desired_count` back to its production value (e.g., 48), and re-deploy. Deploying will increase the number of tasks for SyncGranule back to its desired value, but these won't block DiscoverGranules and QueueGranules since they'll already be running.
+
+## Ingest Monitoring
+
+Monitoring the rate of granule ingest can be done by looking at the SQS queue `backgroundJobQueue` metric "Number Of Messages Deleted", since these correspond 1:1 to granules ingested. Monitoring the progress of a large ingest can be done by looking at the "Approximate Number Of Messages Visible" metric.
```

Comment:
We also need to mention the Cumulus dashboard here for monitoring Granules and Executions. Also the name `IngestAndPublishGranuleWorkflow` might not be standard among the users.